graph-subscriptions-rs/src/subscription_tier.rs

```diff
@@ -0,0 +1,29 @@
+use serde::Deserialize;
+use serde_with::{serde_as, DisplayFromStr};
+
+#[derive(Debug, Default)]
+pub struct SubscriptionTiers(Vec<SubscriptionTier>);
+
+#[serde_as]
+#[derive(Clone, Debug, Default, Deserialize)]
+pub struct SubscriptionTier {
+    /// Payment rate from the subcription contract.
+    #[serde_as(as = "DisplayFromStr")]
+    pub payment_rate: u128,
+    /// Maximum query rate allowed, in queries per second.
+    pub query_rate_limit: u32,
+}
+
+impl SubscriptionTiers {
+    pub fn create(mut tiers: Vec<SubscriptionTier>) -> &'static Self {
+        tiers.sort_by_key(|t| t.payment_rate);
+        Box::leak(Box::new(Self(tiers)))
```

Comment:
We should just return `Self` here. Since this is in a lib now, we shouldn't assume that it's best to do a heap allocation and  leak the memory of that allocation. That only makes sense if we're constructing a small amount of something that remains for the duration of the program execution, which may not be true for all users.

Reply:
Ah okay. I was curious about that. Makes sense, thanks